model/pod/pod.go

```diff
@@ -660,22 +677,26 @@ func (p *Pod) GetSecret() (*Secret, error) {
 }
 
 // SetRunningTask sets the task to dispatch to the pod.
-func (p *Pod) SetRunningTask(ctx context.Context, env evergreen.Environment, taskID string) error {
-	if p.RunningTask == taskID {
+func (p *Pod) SetRunningTask(ctx context.Context, env evergreen.Environment, taskID string, taskExecution int) error {
+	if p.TaskRuntimeInfo.RunningTaskID == taskID && p.TaskRuntimeInfo.RunningTaskExecution == taskExecution {
 		return nil
 	}
-	if p.RunningTask != "" {
-		return errors.Errorf("cannot set running task to '%s' when it is already set to '%s'", taskID, p.RunningTask)
+	if p.TaskRuntimeInfo.RunningTaskID != "" {
+		return errors.Errorf("cannot set running task to '%s' execution %d when it is already set to '%s' execution %d", taskID, taskExecution, p.TaskRuntimeInfo.RunningTaskID, p.TaskRuntimeInfo.RunningTaskExecution)
 	}
 
+	runningTaskIDKey := bsonutil.GetDottedKeyName(TaskRuntimeInfoKey, TaskRuntimeInfoRunningTaskIDKey)
```

Comment:
Can we make these constants in `pod/db.go`?

Reply:
I think we normally put `bsonutil.GetDottedKeyName` for nested key names. I'm not that inclined to add it since people are used to combining dotted key names as they need it for their query.